addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/ConsolePanel.java

```diff
@@ -133,69 +142,112 @@ private boolean isScriptUpdatedOnDisk() {
                 new Thread() {
                     @Override
                     public void run() {
                         this.setName(THREAD_NAME);
                         pollForChanges = true;
+                        if (watchService == null) {
+                            return;
+                        }
+                        WatchKey watchKey;
                         while (pollForChanges) {
-                            if (dialogRunnable == null && isScriptUpdatedOnDisk()) {
-
-                                dialogRunnable =
-                                        () -> {
-                                            String message;
-                                            if (script.isChanged()) {
-                                                message =
-                                                        Constant.messages.getString(
-                                                                "scripts.console.changedOnDiskAndConsole");
-                                            } else {
-                                                message =
-                                                        Constant.messages.getString(
-                                                                "scripts.console.changedOnDisk");
-                                            }
-
-                                            if (JOptionPane.showOptionDialog(
-                                                            ConsolePanel.this,
-                                                            message,
-                                                            Constant.PROGRAM_NAME,
-                                                            JOptionPane.YES_NO_OPTION,
-                                                            JOptionPane.WARNING_MESSAGE,
-                                                            null,
-                                                            SCRIPT_CHANGED_BUTTONS,
-                                                            null)
-                                                    == JOptionPane.YES_OPTION) {
-                                                try {
-                                                    extension.getExtScript().saveScript(script);
-                                                } catch (IOException e) {
-                                                    LOGGER.error(e.getMessage(), e);
+                            try {
+                                watchKey = watchService.take();
+                            } catch (Exception e) {
+                                continue;
+                            }
+                            for (WatchEvent<?> event : watchKey.pollEvents()) {
+                                String changedFileName =
+                                        ((Path) event.context()).getFileName().toString();
+                                if (!changedFileName.equals(script.getName())) {
```

Comment:
The file name and script name might not be always the same, would be better to check the `Path`s instead.